src/coreclr/jit/morph.cpp

```diff
@@ -11909,6 +11909,22 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
             return fgMorphCast(tree);
 
         case GT_MUL:
+            // -a * C => a * -C, where C is constant
+            // MUL(NEG(a), C) => MUL(a, NEG(C))
+            if (fgGlobalMorph && op1->OperIs(GT_NEG) && !op1->gtGetOp1()->IsCnsIntOrI() && op2->IsCnsIntOrI() &&
```

Comment:
`opts.OptimizationEnabled()` check is missed here and in the other cases, when this flag is set to false we don't want to do any CQ transformations, only what is needed for correctness, it is used, for example, in minopts.